packages/dds/tree/src/changeset/delta.ts

```diff
@@ -0,0 +1,291 @@
+ */
+export interface Modify {
+    [type]: typeof MarkType.Modify;
+    [setValue]?: Value;
+    [key: FieldKey]: PositionedMarks;
+}
+
+/**
+ * Describes modifications made to a subtree that is being deleted.
+ */
+export interface ModifyDeleted {
+    [type]: typeof MarkType.Modify;
+    [key: FieldKey]: PositionedMarks<ModifyDeleted | MoveOut>;
+}
+
+/**
+ * Describes modifications made to a subtree that is being moved out.
+ */
+export interface ModifyMovedOut {
+    [type]: typeof MarkType.Modify;
+    [setValue]?: Value;
+    [key: FieldKey]: PositionedMarks<ModifyMovedOut | Delete | MoveOut>;
+}
+
+/**
+ * Describes modifications made to a subtree that is being moved in.
+ */
+export interface ModifyMovedIn {
+    [type]: typeof MarkType.Modify;
+    [key: FieldKey]: PositionedMarks<ModifyMovedIn | MoveIn | Insert>;
+}
+
+/**
+ * Describes modifications made to a subtree that is being inserted.
+ */
+export interface ModifyInserted {
+    [type]: typeof MarkType.Modify;
+    [key: FieldKey]: PositionedMarks<ModifyMovedIn | MoveIn>;
+}
+
+/**
+ * Describes the deletion of a contiguous range of node.
+ * Includes descriptions of the modifications made to those nodes (if any).
+ */
+export interface Delete {
+    [type]: typeof MarkType.Delete;
+    count: number;
+    modify?: PositionedMarks<ModifyDeleted>;
+}
+
+/**
+ * Describes the moving out of a contiguous range of node.
+ * Includes descriptions of the modifications made to those nodes (if any).
+ */
+export interface MoveOut {
+    [type]: typeof MarkType.MoveOut;
+    count: number;
+    /**
+     * The delta should carry exactly one `MoveIn` mark with the same move ID.
+     */
+    moveId: MoveId;
+    modify?: PositionedMarks<ModifyMovedOut>;
+}
+
+/**
+ * Describes the moving in of a contiguous range of node.
+ * Includes descriptions of the modifications made to those nodes (if any).
+ */
+export interface MoveIn {
+    [type]: typeof MarkType.MoveIn;
+    /**
+     * The delta should carry exactly one `MoveOut` mark with the same move ID.
+     */
+    moveId: MoveId;
+    modify?: PositionedMarks<ModifyMovedIn>;
+}
+
+/**
+ * Describes the insertion of a contiguous range of node.
+ * Includes descriptions of the modifications made to those nodes (if any).
+ * Those are represented as `MoveIn` marks within `ProtoField`s.
+ */
+export interface Insert {
+    [type]: typeof MarkType.Insert;
+    content: ProtoNode[];
+    modify?: PositionedMarks<ModifyInserted>;
+}
+
+/**
+ * The contents of a subtree to be created
+ */
+export interface ProtoNode {
+    id: string;
+    type?: string;
+    value?: Value;
+    fields?: ProtoFields;
+}
+
+/**
+ * The fields of a subtree to be created
+ */
+export interface ProtoFields {
+    [key: FieldKey]: ProtoField;
+}
+
+export type ProtoField = ProtoNode[];
+
+export type MoveId = number;
+export type Offset = number;
+export type Index = number;
+export type Value = number | string | boolean | undefined;
+export type NodeId = string;
```

Comment:
It might be good to be clear if this is an Identifier or an Identity, and how it works. Ex: is this a identifier that a node might have on it, or a path which identifies a node? Scope info would be good as well: does this apply to a specific revision? Document? Global? In memory copy of a document?